typescript/cli/src/config/submit.ts

```diff
@@ -0,0 +1,75 @@
+import { stringify as yamlStringify } from 'yaml';
+
+import {
+  PopulatedTransaction,
+  PopulatedTransactionSchema,
+} from '@hyperlane-xyz/sdk';
+import { assert } from '@hyperlane-xyz/utils';
+
+import { WriteCommandContext } from '../context/types.js';
+import { logGray, logRed } from '../logger.js';
+import { getSubmitterBuilder } from '../submit/submit.js';
+import {
+  indentYamlOrJson,
+  readYamlOrJson,
+  writeYamlOrJson,
+} from '../utils/files.js';
+
+export async function runSubmit({
+  context,
+  transactionsFilepath,
+  receiptsFilepath,
+}: {
+  context: WriteCommandContext;
+  transactionsFilepath: string;
+  receiptsFilepath: string;
+}) {
+  const { submissionStrategy, chainMetadata, multiProvider, isDryRun } =
+    context;
+
+  assert(
+    submissionStrategy,
+    'Submission strategy required to submit transactions.\nPlease create a submission strategy, e.g. ./strategy.yaml.',
+  );
+
+  const chain = submissionStrategy.chain;
+  const protocol = chainMetadata[chain].protocol;
+  const submitterBuilder = await getSubmitterBuilder<typeof protocol>({
+    submitterMetadata: submissionStrategy.submitter,
+    transformersMetadata: submissionStrategy.transforms ?? [],
+    multiProvider,
+    chain: submissionStrategy.chain,
+    isDryRun,
+  });
+  const transactions = getTransactions(transactionsFilepath);
+
+  try {
+    const transactionReceipts = await submitterBuilder.submit(...transactions);
+    if (transactionReceipts) {
+      logGray(
+        '🧾 Transaction receipts:\n\n',
+        indentYamlOrJson(yamlStringify(transactionReceipts, null, 2), 4),
+      );
+      writeYamlOrJson(receiptsFilepath, transactionReceipts, 'yaml');
+    }
+  } catch (error) {
+    logRed(
+      `⛔️ Failed to submit ${transactions.length} transactions:`,
+      JSON.stringify(error),
```

Comment:
May be useful to you: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/utils/src/strings.ts#L35

Reply:
sure I can use this. curious when would we ever want less than the entire error message length? the entire stacktrace should be preserved to debug imo

Reply:
It's useful for UIs but maybe not the best option here. Just a suggestion so you know about it